pkg/apis/options/legacy_options.go

```diff
@@ -0,0 +1,98 @@
+package options
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
+)
+
+type LegacyOptions struct {
```

Comment:
Is it clearer to potentially bundle types of legacy options rather than starting 1 big bucket here?

IE for the scope of this PR, we create a set of `LegacyUpstreamOptions` that can be managed together with appropriate helper methods. And then any other things that will be deprecated come V7 structured configs can go into different structs/groupings as it makes sense?

Reply:
(Of course this only makes sense if we flip the struct embedding for the legacy struct to be inside the existing main options struct)

Reply:
I like this idea, I will incorporate this